Tests/UserServiceTests/Validators/UserEmailPasswordValidatorTests.cs

```diff
@@ -0,0 +1,425 @@
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.FirstNameEmpty);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void MissLastName()
+        {
+            var user = new CreateUserRequest
+            {
+                Password = "123",
+                FirstName = "Leo",
+                Birthday = DateTime.Today
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.LastNameEmpty);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void EmailTooLong()
+        {
+            var user = new CreateUserRequest
+            {
+                Email = new string('c', 50) + "@ya.ru",
+                Password = "123",
+                FirstName = "Leo",
+                LastName = "Kor",
+                Birthday = DateTime.Today
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.EmailTooLong);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void FirstNameTooLong()
+        {
+            var user = new CreateUserRequest
+            {
+                Password = "123",
+                FirstName = "C" + new string('c', 40),
+                LastName = "Kor",
+                Birthday = DateTime.Today
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.FirstNameTooLong);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void FirstName1Letter()
+        {
+            var user = new CreateUserRequest
+            {
+                Email = "email@yandex.ru",
+                Password = "123",
+                FirstName = "C",
+                LastName = "Kor",
+                Birthday = DateTime.Today.AddYears(-18)
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.FirstNameTooShort);
+
+            Assert.IsNotNull(expectedError);
+
+        }
+
+
+        [Test]
+        public void Ok()
```

Comment:
Not sure if it's a good name, mb sth like "ValidData" / "ValidDataShouldNotFail" or i don't know

Reply:
fixed